### DISCUSSION TITLE
How to arrange children nodes within a parent node?

### DISCUSSION THREAD

**Discussion by alabhyajindal** [xyflow/xyflow#2572] [Q&A]
![image](https://user-images.githubusercontent.com/52493077/202399379-2eb4c373-6c23-4804-b4d0-d95efd326cb8.png)

I have added a context menu in React Flow which has an option to create a parent. It does two things:
- Creates a new parent node
- Adds `parentNode` property setting it to the newly created parent to all the selected nodes

I wanted help figuring out:
- How to calculate the height and width of the parent so it encompasses all children nodes
- How to calculate the positions of all children nodes so they are arranged in similar way as they were before the parent existed

This is what my solution looks like now. It's achieved by:
- Setting the parent `x, y` to the smallest `x, y` of all the selected nodes
- Fixed parent `width: 300` and `height: 200` 
- Setting children nodes' positions to `x: 0, y: 0`

![image](https://user-images.githubusercontent.com/52493077/202398401-5eca404b-a68b-4c9e-8511-a4da6176ad7f.png)

I would like to achieve the following:

![image](https://user-images.githubusercontent.com/52493077/202399269-3d0d2dd1-a05d-47da-8351-c6a9e90c986f.png)

Thanks in advance. Any help is greatly appreciated!

**Comment by mayurjadhav2002** [comment]
Yes same doubt
